docker-compose.yml

```diff
@@ -351,7 +353,65 @@ services:
       - .:/arrow:delegated
       - ${DOCKER_VOLUME_PREFIX}ubuntu-ccache:/ccache:delegated
     command: *cpp-command
+  
+  ubuntu-cpp-static:
+    # Usage:
+    #   docker-compose build ubuntu-cpp-static
+    #   docker-compose run --rm ubuntu-cpp-static
+    # Parameters:
+    #   ARCH: amd64, arm64v8, s390x, ...
+    #   UBUNTU: 18.04, 20.04
+    image: ${REPO}:${ARCH}-ubuntu-${UBUNTU}-cpp-static
+    build:
+      context: .
+      dockerfile: ci/docker/ubuntu-${UBUNTU}-cpp.dockerfile
+      cache_from:
+        - ${REPO}:${ARCH}-ubuntu-${UBUNTU}-cpp-static
+      args:
+        arch: ${ARCH}
+        base: "${ARCH}/ubuntu:${UBUNTU}"
+        clang_tools: ${CLANG_TOOLS}
+        llvm: ${LLVM}
+        gcc_version: ${GCC_VERSION}
+    shm_size: *shm-size
+    cap_add:
+      - SYS_ADMIN
+    devices:
+      - "/dev/fuse:/dev/fuse"
+    security_opt:
+      - apparmor:unconfined
+    ulimits: *ulimits
+    environment:
+      <<: *ccache
+      ARROW_HOME: /arrow
+      ARROW_DEPENDENCY_SOURCE: BUNDLED
+      LIBARROW_MINIMAL: "false"
+      ARROW_MIMALLOC: "ON"
+    volumes: *ubuntu-volumes
+    command: /bin/bash -c "
+        cd /arrow && r/inst/build_arrow_static.sh"
 
+  centos-cpp-static:
```

Comment:
If we would like to keep these builds runnable locally without archery commands then we should add a presets feature to `archery docker run` and keep docker-compose.yml mostly as a template. 
Ideally we should move away from docker-compose at some point and either use our own implementation (e.g. [inspiration from ursabot](https://github.com/ursa-labs/ursabot/blob/master/ursabot/docker.py)) or preferably https://earthly.dev with better code reuse.